default/AI/InvasionAI.py

```diff
@@ -394,112 +394,113 @@ def evaluate_invasion_planet(planetID, missionType, fleetSupplyablePlanetIDs, em
-        if not theseFleets:
-            if not FleetUtilsAI.stats_meet_reqs(foundStats, minStats):
-                print "Insufficient invasion troop allocation for system %d ( %s ) -- requested %s , found %s"%(sysID, universe.getSystem(sysID).name, minStats, foundStats)
-                invasionPool.update( foundFleets )
+    universe = fo.getUniverse()
+    invasion_fleet_pool = set(fleet_ids)
+    if not invasion_fleet_pool:
+        return
+
+    for pID, pscore, ptroops in evaluated_planets:
+        planet = universe.getPlanet(pID)
+        if not planet:
+            continue
+        sys_id = planet.systemID
+        found_fleets = []
+        found_stats = {}
+        min_stats = {'rating': 0, 'troopCapacity': ptroops}
+        target_stats = {'rating': 10, 'troopCapacity': ptroops+1}
+        these_fleets = FleetUtilsAI.get_fleets_for_mission(1, target_stats, min_stats, found_stats, "",
+                                                           systems_to_check=[sys_id], systems_checked=[],
+                                                           fleet_pool_set=invasion_fleet_pool, fleet_list=found_fleets,
+                                                           verbose=False)
+        if not these_fleets:
+            if not FleetUtilsAI.stats_meet_reqs(found_stats, min_stats):
+                print "Insufficient invasion troop allocation for system %d ( %s ) -- requested %s , found %s" % (
+                    sys_id, universe.getSystem(sys_id).name, min_stats, found_stats)
+                invasion_fleet_pool.update(found_fleets)
                 continue
             else:
-                theseFleets = foundFleets
-        aiTarget = AITarget.AITarget(EnumsAI.TargetType.TARGET_PLANET, pID)
-        print "assigning invasion fleets %s to target %s"%(theseFleets, aiTarget)
-        for fleetID in theseFleets:
-            fleet=universe.getFleet(fleetID)
-            aiFleetMission = foAI.foAIstate.get_fleet_mission(fleetID)
-            aiFleetMission.clear_fleet_orders()
-            aiFleetMission.clear_targets( (aiFleetMission.get_mission_types() + [-1])[0] )
-            aiFleetMission.add_target(missionType, aiTarget)
+                these_fleets = found_fleets
+        target = AITarget.AITarget(EnumsAI.TargetType.TARGET_PLANET, pID)
+        print "assigning invasion fleets %s to target %s" % (these_fleets, target)
+        for fleetID in these_fleets:
+            fleet_mission = foAI.foAIstate.get_fleet_mission(fleetID)
+            fleet_mission.clear_fleet_orders()
+            fleet_mission.clear_targets((fleet_mission.get_mission_types() + [-1])[0])
+            fleet_mission.add_target(mission_type, target)
 
 
 def assign_invasion_fleets_to_invade():
-    # assign fleet targets to invadable planets
-    
+    """assign fleet targets to invadable planets"""
```

Comment:
The docstring is a phrase ending in a period. [PEP257](https://www.python.org/dev/peps/pep-0257/)